From a47232facc446039ed509100f80ebb7de621fffa Mon Sep 17 00:00:00 2001
Message-ID: <a47232facc446039ed509100f80ebb7de621fffa.1749113303.git.jdenemar@redhat.com>
From: Collin Walling <walling@linux.ibm.com>
Date: Mon, 16 Dec 2024 18:03:55 -0500
Subject: [PATCH] libvirt-domain: introduce
 VIR_CONNECT_GET_DOMAIN_CAPABILITIES_DISABLE_DEPRECATED_FEATURES

Introduce domain flag used to filter deprecated features from the
domain's CPU model.

Signed-off-by: Collin Walling <walling@linux.ibm.com>
Reviewed-by: Jiri Denemark <jdenemar@redhat.com>
(cherry picked from commit 4e2c8de2047e21d98443944a2bfe94529b269efa)
JIRA: https://issues.redhat.com/browse/RHEL-88716
Signed-off-by: Thomas Huth <thuth@redhat.com>
---
 include/libvirt/libvirt-domain.h | 12 ++++++++++++
 src/libvirt-domain.c             |  2 +-
 2 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/include/libvirt/libvirt-domain.h b/include/libvirt/libvirt-domain.h
index 792973ce2d..d3101b112b 100644
--- a/include/libvirt/libvirt-domain.h
+++ b/include/libvirt/libvirt-domain.h
@@ -1160,6 +1160,18 @@ int virDomainMigrateGetMaxSpeed(virDomainPtr domain,
 int virDomainMigrateStartPostCopy(virDomainPtr domain,
                                   unsigned int flags);
 
+/**
+ * virConnectGetDomainCapabilitiesFlags:
+ *
+ * Domain capabilities flags.
+ *
+ * Since: 11.0.0
+ */
+typedef enum {
+    /* Report host model with deprecated features disabled. (Since: 11.0.0) */
+    VIR_CONNECT_GET_DOMAIN_CAPABILITIES_DISABLE_DEPRECATED_FEATURES = (1 << 0),
+} virConnectGetDomainCapabilitiesFlags;
+
 char * virConnectGetDomainCapabilities(virConnectPtr conn,
                                        const char *emulatorbin,
                                        const char *arch,
diff --git a/src/libvirt-domain.c b/src/libvirt-domain.c
index 5912551a49..7083f10f04 100644
--- a/src/libvirt-domain.c
+++ b/src/libvirt-domain.c
@@ -11573,7 +11573,7 @@ virDomainSetUserPassword(virDomainPtr dom,
  * @arch: domain architecture
  * @machine: machine type
  * @virttype: virtualization type
- * @flags: extra flags; not used yet, so callers should always pass 0
+ * @flags: extra flags; bitwise-OR of virConnectGetDomainCapabilitiesFlags
  *
  * Prior creating a domain (for instance via virDomainCreateXML
  * or virDomainDefineXML) it may be suitable to know what the
-- 
2.49.0
